NitroxPatcher/Patches/Dynamic/uGUI_SignInput_OnDeselect_Patch.cs

```diff
@@ -18,9 +18,23 @@ public static void Postfix(uGUI_SignInput __instance)
         {
             GameObject gameObject = __instance.gameObject.FindAncestor<PrefabIdentifier>().gameObject;
             NitroxId id = NitroxEntity.GetId(gameObject);
-
+            TechTag tag = gameObject.GetComponent<TechTag>();
             SignMetadata signMetadata = new(__instance.text, __instance.colorIndex, __instance.scaleIndex, __instance.elementsState, __instance.IsBackground());
-            NitroxServiceLocator.LocateService<Building>().MetadataChanged(id, signMetadata);
+            
+            switch (tag.type)
+            {
+                case TechType.SmallStorage:
+                    // In the water
+                    Resolve<Entities>().BroadcastMetadataUpdate(id, EntitySignMetadata.FromSignMetadata(signMetadata));
```

Comment:
Instead of creating the `SignMetadata` and then changing it to `EntitySignMetadata` we can create the corresponding metadata directly in this switch statement. Then we also don't need `FromSignMetadata()`.